internal/jennies/java/converter.go

```diff
@@ -0,0 +1,167 @@
+package java
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/grafana/codejen"
+	"github.com/grafana/cog/internal/ast"
+	"github.com/grafana/cog/internal/jennies/common"
+	"github.com/grafana/cog/internal/jennies/template"
+	"github.com/grafana/cog/internal/languages"
+	"github.com/grafana/cog/internal/orderedmap"
+	"github.com/grafana/cog/internal/tools"
+)
+
+const variableModel = "VariableModel"
+
+type Converter struct {
+	config         Config
+	nullableConfig languages.NullableConfig
+	tmpl           *template.Template
+}
+
+func (jenny *Converter) JennyName() string {
+	return "JavaConverter"
+}
+
+func (jenny *Converter) Generate(context languages.Context) (codejen.Files, error) {
+	files := codejen.Files{}
+
+	for _, builder := range context.Builders {
+		output, err := jenny.generateConverter(context, builder)
+		if err != nil {
+			return nil, err
+		}
+
+		filename := filepath.Join(
+			jenny.config.ProjectPath,
+			formatPackageName(builder.Package),
+			fmt.Sprintf("%sConverter.java", tools.UpperCamelCase(builder.Name)),
```

Comment:
Nitpick (doesn't need to get fixed in this PR): we should really use functions like `formatPackageName()`, `formatObjectName()`, ... everywhere